spec/rubocop/cop/rspec/pending_without_reason_spec.rb

```diff
@@ -136,6 +173,21 @@
     end
   end
 
+  context 'when pending/skip inside conditional' do
+    it 'registers offense' do
+      expect_offense(<<~RUBY)
+        it 'does something' do
+          pending if RUBY_VERSION < '3.0'
+          ^^^^^^^ Give the reason for pending.
```

Comment:
This is a point I have concerns involving existing behavior. I think this cop can allow it when `pending` (and `skip`) in a condition. For example, `Minitest/SkipWithoutReason` cop allows this case. What do you think?
https://github.com/rubocop/rubocop-minitest/issues/198

Reply:
I would say that in cases like `pending if RUBY_VERSION < '3.0'` or `pending if $testing_new_activerecord_version` or `pending if new_version?` it’s pretty obvious why the spec is pending. I think we shouldn’t need to enforce adding a message in these cases.

Reply:
Indeed, it is. We have modified the if/unless case so that it is not an offenses.